[AERIE-1996] feat: convert all <Table /> components to <DataGrid /> components

duranb opened the pull request:
This resolves https://jira.jpl.nasa.gov/browse/AERIE-1996

The views contain the remaining tables that have been converted are the following:
- Dictionaries
- ExpansionPanel
- ExpansionRules
- ExpansionSets
- Models
- Plans
- ViewsPanel

duranb committed:
feat: convert all <Table /> components to <DataGrid /> components
remove <Table /> component from project

duranb committed:
fix typo in buttons ExpansionRules table buttons
fix tests

camargo commented on src/components/activity/ActivityTable.svelte:
```diff
@@ -24,8 +24,17 @@
   rowData={$activities}
-  on:rowSelected={({ detail }) => ($selectedActivityId = detail.data.id)}
+  selectedRowIds={$selectedActivityIds}
+  on:rowSelected={({
```
Can we put this in it's own function just to clean up the template a bit?

duranb committed:
add property to disable row highlight upon selection
disable row selection highlight for model and plan tables
remove unused table types

duranb committed:
separate out to named function
fix selection loop bug in expansion sets
fix DataGrid row selection reactive logic

duranb committed:
lighten table border
dispatch cell context menu event
